File: NeetCode/Arrays/Easy/1299.ReplaceElementsWithGreatest.py
#Given an array arr, replace every element in that array 
#with the greatest element among the elements to its right, 
#and replace the last element with -1.

def replaceElements(arr):
    maximum = -1

    for number in range(len(arr) - 1, -1, -1):
        old_max = maximum

        if arr[number] > old_max:
            maximum = arr[number]

        arr[number] = old_max

    return arr

print(replaceElements([17,18,5,4,6,1]))
print(replaceElements([400]))

File: NeetCode/Swift/Arrays/Easy/1299.ReplaceElementsWithGreatest.swift
//Given an array arr, replace every element in that array 
//with the greatest element among the elements to its right, 
//and replace the last element with -1.

func replaceElements(_ arr: [Int]) -> [Int] {
    var arr = arr
    var greatest = -1
    
    for i in stride(from: arr.count - 1, to: -1, by: -1) {
        var current = arr[i]
        arr[i] = greatest
        greatest = max(greatest, current)
    }

    return arr
}

print(replaceElements([17,18,5,4,6,1]))
print(replaceElements([400]))